### DISCUSSION TITLE
Fasthtml vs reflex

### DISCUSSION THREAD

**Discussion by j209** [AnswerDotAI/fasthtml#391] [General]
Whats difference between two frameworks:
Fasthtml & [Reflex](https://github.com/reflex-dev/reflex)

**Comment by ijohn** [comment]
A couple of things from the top of my head:

- FastHTML uses HTMX, while Reflex uses React, for the frontend library
- There will be 2 servers running for Reflex: Next.js to serve frontend, and FastAPI to serve the backend. FastHTML AFAIK only runs 1 server, FastAPI

**Reply by vacmar01** [reply]
Reflex compiles everything down to a React Frontend and a FastAPI backend with json based communication between the two parts.

FastHTML sends HTML from the server to the browser. Nothing gets compiled or anything.

**Reply by jamilraichouni** [reply]
> A couple of things from the top of my head:
> 
> * FastHTML uses HTMX, while Reflex uses React, for the frontend library
> * There will be 2 servers running for Reflex: Next.js to serve frontend, and FastAPI to serve the backend. FastHTML AFAIK only runs 1 server, FastAPI

Under the hood, FastHTML runs `starlette` (https://github.com/encode/starlette) and not FastAPI. You can see this in the list of dependencies:

https://github.com/AnswerDotAI/fasthtml/blob/6abe003e4466f6aa504759fd7036731848a7b8a3/settings.ini#L7

**Comment by ToufiqQureshi** [comment]
i think fasthtml is best beacouse have alot of tutorials so it make this easy to debug or any kind of help but reflex doesnt have any single tutorials so thath why '
